homeassistant/components/netgear/device_tracker.py

```diff
@@ -19,19 +20,42 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up device tracker for Netgear component."""
+    router = hass.data[DOMAIN][entry.unique_id][KEY_ROUTER]
+    coordinator = hass.data[DOMAIN][entry.unique_id][KEY_COORDINATOR]
+    tracked = set()
 
-    def generate_classes(router: NetgearRouter, device: dict):
-        return [NetgearScannerEntity(router, device)]
+    @callback
+    def new_device_callback() -> None:
+        """Add new devices if needed."""
+        if not coordinator.data:
+            return
+
+        new_entities = []
+
+        for mac, device in router.devices.items():
+            if mac in tracked:
+                continue
+
+            new_entities.append(NetgearScannerEntity(coordinator, router, device))
+            tracked.add(mac)
 
-    async_setup_netgear_entry(hass, entry, async_add_entities, generate_classes)
+        if new_entities:
+            async_add_entities(new_entities, update_before_add=True)
```

Comment:
We don't need to refresh the coordinator again.
```suggestion
            async_add_entities(new_entities)
```